GitLab/src/stats.js

```diff
@@ -0,0 +1,107 @@
+// jshint esversion: 9
+
+/**
+ * @description null
+ * @param {ParamsType} params list of command parameters
+ * @param {?string} commandText text message
+ * @param {!object} [secrets = {}] list of secrets
+ * @return {Promise<SlackBodyType>} Response body
+ */
+
+var XMLHttpRequest;
+
+async function install(pkgs) {
+  
+  pkgs = pkgs.join(' ');
+  return new Promise((resolve, reject) => {
+    const { exec } = require('child_process');
+    exec(`npm install ${pkgs}`, (err, stdout, stderr) => {
+      if (err) reject(err);
+      else resolve();
+    });
+  });
+}
+
+// Performs GET requests on provided url and returns results as JSON
+function fetchStats(url) {
+  
+  XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+  var xmlHttp = new XMLHttpRequest();
+  xmlHttp.open( "GET", url, false );
+  xmlHttp.send( null );
+  return JSON.parse(xmlHttp.responseText);
+}
+
+async function _command(params, commandText, secrets = {}) {
+  
+  if (!XMLHttpRequest) {
+    let packages = [ 'xmlhttprequest' ];
+    await install(packages);
+    XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+  }
+  const {

```

Comment:
Probably should be a check to see if the secret is present?

Reply:
Stats does not use a secret 😅

Reply:
!